gitpod-io/gitpod
# [dashboard] Open Workspaces in New Tab and Redirect to Dashboard

## Description


This PR implements a new behavior for opening workspaces from the dashboard:

- When a user creates a new workspace or opens an existing one from /workspaces, it now opens in a new tab.
- After initiating a new workspace, the user is redirected to /workspaces in the current tab.


These changes aim to improve user experience by:

- Maintaining the user's connection to the Gitpod dashboard
- Enhancing discoverability of Gitpod's educational components
- Providing easier navigation back to Gitpod for users who close their workspace

This update particularly benefits new users, especially those coming directly from a Git context, by ensuring they have continued access to Gitpod's full feature set and educational resources.

## Related Issue(s)

Fixes OPM-680

## How to test


- Open & Complete onboarding in preview environment
- Create a new workspace and see the new flow 👀 (more details in linked linear issue)
- Go back to `/workspaces`, click on workspace id and see that link would open in new tab. And, also when you click on three vertical dots and then `Open`, it would open the workspace in new tab

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - siddhant-of11c389904</li>
	<li><b>🔗 URL</b> - <a href="https://siddhant-of11c389904.preview.gitpod-dev.com/workspaces" target="_blank">siddhant-of11c389904.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - siddhant-opm-680-workspaces-should-be-open-in-a-new-tab-when-started-gha.27602</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-siddhant-of11c389904%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
